utils/anki_connect.py

```diff
@@ -49,35 +46,40 @@ def __cast_raw_card_to_Card(self, raw_card_data):
             )
             return my_Card
         except:
-            return None
+            raise KeyError("Cannot create Card")
+            
         
 
     def get_Cards_from_deck(self, deck_name) -> list[Card]:
        
-        cards_raw_data = self.__get_cards_data(deck_name)
+        cards_raw_data = self.get_cards_data(deck_name)
+
         deck_of_Cards = []
-        for i in cards_raw_data:
-            deck_of_Cards.append(self.__cast_raw_card_to_Card(i))
+
+        for c in cards_raw_data:
+            try:
+                deck_of_Cards.append(self.cast_raw_card_to_Card(c))
+            except:
+                try:
+                    print(f"Failed at {c['cardId']}")
+                except:
+                    print(f"Corrupted card skipped: {c}")
+        print(len(cards_raw_data), len(deck_of_Cards))
```

Comment:
A good pattern here is:

```python
try:
    do_something()
except Exception as e:
    id = card.get("cardId", "corrupted card")
    print(f"Failed at {id}: exception is {e}")
```

But this is just nitpicking -- current state works well!